src/main/java/core/basesyntax/RightTriangleSupplier.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class RightTriangleSupplier {
+    private Random random = new Random();
+    private RandomSize size = new RandomSize();
+    private ColorSupplier color = new ColorSupplier();
+
+    public Figure getRandomRightTriangle() {
+        return new RightTriangle(size.getRandomSize(),
+                size.getRandomSize(), color.getRandomColor());

```

Comment:
Creating a figure, it's better not to pass expressions directly in the constructor for better code readability. Consider creating separate variables for the sizes and color before passing them to the constructor of `RightTriangle`.